frontend/pages/ManageControlsPage/Secrets/components/DeleteSecretModal/DeleteSecretModal.tsx

```diff
@@ -44,9 +44,9 @@ const DeleteSecretModal = ({
           ? errorObject.base
           : "An error occurred while deleting the custom variable. Please try again.";
       renderFlash("error", message);
+      onExit();
     } finally {
       setIsDeleting(false);
-      onExit();
```

Comment:
Either `onDeleteSecret` or `onExit` will be called, and both of them close the modal, so no need to do it in `finally`.